graylog2-web-interface/src/components/event-definitions/event-definition-types/AggregationConditionExpressions/NumberRefExpression.jsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import lodash from 'lodash';
+import naturalSort from 'javascript-natural-sort';
+
+import { Select } from 'components/common';
+import { Col, ControlLabel, FormGroup, HelpBlock, Row } from 'components/graylog';
+
+import { numberRefNodePropType } from 'logic/alerts/AggregationExpressionTypes';
+
+const formatFunctions = (functions) => {
+  return functions
+    .sort(naturalSort)
+    .map(fn => ({ label: `${fn.toLowerCase()}()`, value: fn }));
+};
+
+const NumberRefExpression = ({
+  aggregationFunctions,
+  formattedFields,
+  eventDefinition,
+  expression,
+  onChange,
+  renderLabel,
+  validation,
+}) => {
+  const getSeries = (seriesId) => {
+    return eventDefinition.config.series.find(series => series.id === seriesId);
+  };
+
+  const createSeries = () => {
+    return { id: expression.ref };
+  };
+
+  const getOrCreateSeries = (seriesId) => {
+    return getSeries(seriesId) || createSeries();
+  };
+
+  const getSeriesId = (currentSeries, func, field) => {
+    return `${lodash.defaultTo(func, currentSeries.function)}-${lodash.defaultTo(field, currentSeries.field || '')}`;
```

Comment:
Hmm, this seems to be broken.
If I add a new condition to an already existing count() I get this json:
```
 "series" : [ 
            {
                "id" : "31bf89e7-33d7-49bd-8e16-ad5160867b9a",
                "function" : "count",
                "field" : null
            }, 
            {
                "id" : "count-",
                "function" : "count",
                "field" : null
            }
```
The new Id is called `count-` which also leads to the series being added twice.

Reply:
This change is to avoid exactly that problem to happen. I don't think the frontend is the right place to normalize existing series, but we can think of adding a migration in the server for that.

Reply:
So, to clarify, you get 2 series because one of them was already added and has a unique ID. Modifying that series in the frontend will change the ID to `<function>-<field>`, and since that's already there, there will only be one series.

Reply:
Ok, so that's on purpose.
So without the migration the backend would need to compute the series twice?

Reply:
It may be, depends on what the user does. Computing the series twice doesn't work in the backend at the moment and I don't know how to fix it. I think it should also not be done, since that will only return the same results twice.

Reply:
So from my point of view we should either do a migration or document this and ask people to edit the original condition if they run in this issue. I think the first option is a bit better, but I also didn't include it in the PR since I'm unsure how big of a problem this will be.

Reply:
I've added a migration to this PR.